geoarrow-types/src/geoarrow/types/constants.py

```diff
@@ -162,8 +162,11 @@ class Dimensions(TypeSpecEnum):
     <Dimensions.XYZM: 4>
     """
 
-    UNSPECIFIED = 0
-    """Unknown or uninitialized dimensions"""
+    UNSPECIFIED = -1
+    """Unspecified dimensions"""
+
+    UNKNOWN = 0
+    """Unknown or mixed dimensions"""
```

Comment:
I'm not 100% clear on the difference between unspecified and unknown

Reply:
As an aside, I do like the ideas of `coalesce` and `common` for types. I had so much pain working with type casting in geoarrow-rs